Dfe.Academies.Academisation.Service/Commands/Legacy/Project/LegacyProjectUpdateCommand.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Dfe.Academies.Academisation.Core;
+using Dfe.Academies.Academisation.IData.ProjectAggregate;
+using Dfe.Academies.Academisation.IService.Commands.Project;
+using Dfe.Academies.Academisation.IService.ServiceModels.Legacy.ProjectAggregate;
+using Dfe.Academies.Academisation.Service.Mappers.Legacy.ProjectAggregate;
+
+namespace Dfe.Academies.Academisation.Service.Commands.Project;
+
+public class LegacyProjectUpdateCommand : ILegacyProjectUpdateCommand
+{
+	private readonly IProjectGetDataQuery _projectGetDataQuery;
+	private readonly IProjectUpdateDataCommand _projectUpdateDataCommand;
+
+	public LegacyProjectUpdateCommand(IProjectGetDataQuery projectGetDataQuery, IProjectUpdateDataCommand projectUpdateDataCommand)
+	{
+		_projectGetDataQuery = projectGetDataQuery;
+		_projectUpdateDataCommand = projectUpdateDataCommand;
+	}
+
+	public async Task<CommandResult> Execute(LegacyProjectServiceModel legacyProjectServiceModel)
+	{
+		var existingProject = await _projectGetDataQuery.Execute(legacyProjectServiceModel.Id);
+
+		if (existingProject is null)
+		{
+			return new NotFoundCommandResult();
+		}
+
+		var result = existingProject.UpdatePatch(legacyProjectServiceModel.ToDomain());
+				
```

Comment:
Should guard against failure results returned from the domain here (I know we don't have any yet but this class doesn't know that, at least it should throw a NotImplementedException to be a placeholder)

Reply:
Done